src/dune_pkg/opam_repo.ml

```diff
@@ -154,10 +154,10 @@ let load_packages_from_git rev_store opam_packages =
         ~files_dir:(Some files_dir))
 ;;
 
-let all_packages_versions_in_dir loc ~dir opam_package_name =
-  let dir = Path.append_local dir (Paths.package_root opam_package_name) in
+let all_packages_versions_in_dir_at_path ~f ~dir ~path loc =
```

Comment:
The `f` here is rather confusing, as it is not clear what it would be applied on.

Reply:
I've improved this, thanks!